src/test/java/htsjdk/samtools/CRAMContainerStreamWriterTest.java

```diff
@@ -182,4 +204,43 @@ public void testCRAMContainerStreamWithIndex() throws IOException {
         Assert.assertEquals(count, 2);
     }
 
+    @Test(description = "Test CRAMContainerStream with crai index")
+    public void testCRAMContainerStreamWithCraiIndex() throws IOException {
+        final List<SAMRecord> samRecords = createRecords(100);
+        final SAMFileHeader header = createSAMHeader(SAMFileHeader.SortOrder.coordinate);
+        final ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+        final ByteArrayOutputStream indexStream = new ByteArrayOutputStream();
+        doTest(samRecords, outStream, header, new CRAMCRAIIndexer(indexStream, header));
+        outStream.close();
+        indexStream.close();
+
+        // write the file out
+        final File cramTempFile = File.createTempFile("cramContainerStreamTest", ".cram");
+        cramTempFile.deleteOnExit();
+        final OutputStream cramFileStream = new FileOutputStream(cramTempFile);
+        cramFileStream.write(outStream.toByteArray());
+        cramFileStream.close();
+
+        // write the index out
+        final File indexTempFile = File.createTempFile("cramContainerStreamTest", ".crai");
+        indexTempFile.deleteOnExit();
+        OutputStream indexFileStream = new FileOutputStream(indexTempFile);
+        indexFileStream.write(indexStream.toByteArray());
+        indexFileStream.close();
+
+        final ReferenceSource refSource = createReferenceSource();
+        final CRAMFileReader reader = new CRAMFileReader(
+                cramTempFile,
+                indexTempFile,
+                refSource,
+                ValidationStringency.SILENT);
+        final CloseableIterator<SAMRecord> iterator = reader.query(new QueryInterval[]{new QueryInterval(1, 10, 10)}, false);
+        int count = 0;
+        while (iterator.hasNext()) {
+            SAMRecord actualRecord = iterator.next();
+            count++;
+        }
+        Assert.assertEquals(count, 2);
+    }
+
```

Comment:
This whole block is a duplicate of the code in `testCRAMContainerStreamWithBaiIndex`, and can be extracted into a method with a file extension paramater.